walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,104 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.lang.Math;
 
 public final class FindMeetingQuery {
+
+  private ArrayList<TimeRange> meetings = new ArrayList<TimeRange>();
+  private ArrayList<TimeRange> timeTable = new ArrayList<TimeRange>();
+  private int meetingDuration;
+  private Collection<String> attendees;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    meetingDuration = (int) request.getDuration();
+    attendees = request.getAttendees();
+    
+    // Edge Cases
+    // 1. when requested duration exceeds one day 
+    // 2. there are no attendees requested to the meeting
+    if (request.getDuration() > (60*24)) {
+        return Arrays.asList();
+    } else if (request.getAttendees().size() == 0) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    constructTimeTable(events);
+
+    // when there are no timeslots where attendees need to attend events
+    if (timeTable.size() == 0) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    
+    findMeetingSlots(timeTable);
+
+    return meetings;
+  }
+  
+  // aggregates a timetable based on the schedule of attendees involved 
+  public void constructTimeTable(Collection<Event> events) {
```

Comment:
Keep methods private by default unless they are needed outside. It's preferable to not expose implementation detail, so that other people cannot depend on it. In case you need to change implementation, it will be easier to change without worrying about external dependencies.

Same for the methods below